src/main/kotlin/org/kamiblue/client/plugin/PluginInfo.kt

```diff
@@ -0,0 +1,84 @@
+package org.kamiblue.client.plugin
+
+import com.google.gson.Gson
+import com.google.gson.annotations.SerializedName
+import org.kamiblue.client.plugin.api.Plugin
+import org.kamiblue.commons.interfaces.Nameable
+import java.io.InputStream
+
+class PluginInfo private constructor(
+    @SerializedName("name") private val name0: String?,
+    @SerializedName("version") private val version0: String?,
+    @SerializedName("authors") private val authors0: Array<String>?,
+    @SerializedName("description") private val description0: String?,
+    @SerializedName("url") private val url0: String?,
+    @SerializedName("kami_version") private val kamiVersion0: String?,
+    @SerializedName("required_plugins") private val requiredPlugins0: Array<String>?,
+    @SerializedName("main_class") private val mainClass0: String?,
+    @SerializedName("hot_reload") private val hotReload0: Boolean?
+) : Nameable {
+
+    /** The name of the plugin, will be used as both an identifier and a display name */
+    override val name: String get() = name0.nonBlank("name")
+
+    /** The plugin's version */
+    val version: String get() = version0.nonBlank("version")
+
+    /** A list of the names of the plugin's authors */
+    val authors: Array<String> get() = authors0 ?: authorsNull
+
+    /** A short description of the plugin */
+    val description: String get() = description0 ?: descriptionNull
+
+    /** A link to the plugin's website */
+    val url get() = url0 ?: urlNull
+
+    /** The minimum version of KAMI Blue required for the plugin to run. */
+    val kamiVersion: String get() = kamiVersion0.nonBlank("kami_version")
```

Comment:
I won't lie; I'm still iffy about this KAMI version business.

I think min kami version or something of that sort would be better; KAMI version alone implies that the plugin can only be used on this version only.